src/main/java/com/modern/training/platform/config/MetricsConfig.java

```diff
@@ -0,0 +1,77 @@
+package com.modern.training.platform.config;
+
+import java.net.DatagramSocket;
+import java.net.InetAddress;
+import java.net.SocketException;
+import java.net.UnknownHostException;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.actuate.autoconfigure.metrics.MeterRegistryCustomizer;
+import org.springframework.context.annotation.Configuration;
+
+import io.micrometer.core.instrument.MeterRegistry;
+//import lombok.extern.slf4j.Slf4j;
+
+@Configuration
+//@Slf4j
+public class MetricsConfig {
+	
+	Logger log = LoggerFactory.getLogger(this.getClass());
+
+  	private static final String PROFILE_TAG_KEY = "profile";
+  	private static final String APP_TAG_KEY = "appname";
+	private static final String HOST_TAG_KEY = "host";
+  	private static final String IP_TAG_KEY = "ip";
+  
+  	@Value("${spring.application.name}")
+  	private String appName;
+
+  	@Value("${spring.profile:dev}")
+  	private String profile;
+    
+
+  	public MeterRegistryCustomizer<MeterRegistry> customizeRegistry() {
+    
+      return registry -> {
+        registry.config()
+        	.commonTags(APP_TAG_KEY, appName)
+        	.commonTags(PROFILE_TAG_KEY, profile)
+       		.commonTags(HOST_TAG_KEY, getHostname())
+       		.commonTags(IP_TAG_KEY, getIpAddress())
+//	        .meterFilter(MeterFilter.ignoreTags("too.much.information"))
+//	        .meterFilter(MeterFilter.denyNameStartsWith("jvm"))
+	        ;
+      };
+    }
+    
+    private String getHostname() {
+    	
+	   String hostname = "Unknown";
+
+	    try {
+	        InetAddress addr;
+	        addr = InetAddress.getLocalHost();
+	        hostname = addr.getHostName();
+	    } catch (UnknownHostException e) {
+	        log.error("Hostname can not be resolved", e);
+	    }
+	    
+	    return hostname;
+    }
+    
+    private String getIpAddress() {
+
+    	String ip = "Unknown";
```

Comment:
Why do we initailze the Strings IP and Hostname with Unknown instead of like Empty String and If we are doing maybe we can have a constant UNKNOWN = "Unknown". So we can use the consistency if anywhere doing any equal checks.

Reply:
The tag needs a value for Prometheus. Also, this config was meant as an example too - I thought we'd discuss in the meeting and decide what we actually need.